specs/protocol/block_header.md

```diff
@@ -0,0 +1,24 @@
+# Block Header
+
+## Application Header
+
+name                 | type       | description
+---------------------|------------|-----------------------------------------------------------------------------------------------------
+`height`             | `uin64`    | Height of this block.
+`timestamp`          | `uin64`    | Time this block was created, in [TAI64](https://cr.yp.to/libtai/tai64.html) format.
+`txCount`            | `uint64`   | Number of [transaction](./tx_format.md#transaction)s in this block.
+`txRoot`             | `byte[32]` | [Merkle root](./cryptographic_primitives.md#binary-merkle-tree) of [transaction](./tx_format.md#transaction)s in this block.
+`outputMessagesCount`|`uint64`| Number of [output message](./tx_format.md#outputmessage)s in this block.
+`outputMessagesRoot` | `byte[32]` | [Merkle root](./cryptographic_primitives.md#binary-merkle-tree) of [output message](./tx_format.md#outputmessage)s in this block.
+
+## Consensus Header
+
+name                   | type       | description
+-----------------------|------------|-----------------------------------------------------------------------------------------------------------------------
+`prevRoot`           | `byte[32]` | [Merkle root](./cryptographic_primitives.md#binary-merkle-tree) of all previous [consensus header](#consensus-header) hashes (i.e. not including this block).
+`applicationHash`      | `byte[32]` | [Hash](./cryptographic_primitives.md#hashing) of serialized [application header](#application-header).
+`nextValidatorSetRoot` | `byte[32]` | [Merkle root](./cryptographic_primitives.md#binary-merkle-tree) of the validator set that will vote on the next block.
+`nextVotingPower`      | `uint64`   | Voting power of the validator set that will vote on the next block.
+`prevCommitHash`       | `byte[32]` | [Hash](./cryptographic_primitives.md#hashing) of consensus commit for `prevRoot`.
```

Comment:
Is it overkill to make votes merkelized so we can efficiently prove that a given vote is in the set of votes that defined this block?

Reply:
It's not overkill, but there's also not really any usecase for that. The contract verifies all votes as they come in, so anything after that should just be as cheap as possible.